tests/tests/tier0/bluechi-agent-config-c-option/test_agent_config_c_option.py

```diff
@@ -0,0 +1,63 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import os
+import logging
+
+from typing import Dict
+from bluechi_test.test import BluechiTest
+from bluechi_test.container import BluechiControllerContainer, BluechiNodeContainer
+from bluechi_test.config import BluechiControllerConfig, BluechiNodeConfig
+from bluechi_test.util import read_file
+
+LOGGER = logging.getLogger(__name__)
+
+NODE_FOO = "node-foo"
+active_status = "active"
+failed_status = "failed"
+
+
+def restart_with_config_file(config_file_location, container, service):
+    container.exec_run(f"sed -i '/ExecStart=/c\\ExecStart=/usr/libexec/{service} -c "
+                       f"{config_file_location}' "
+                       f"/usr/lib/systemd/system/{service}.service")
+    container.exec_run("systemctl daemon-reload")
+    container.exec_run(f"systemctl restart {service}.service")
+
+
+def exec(ctrl: BluechiControllerContainer, nodes: Dict[str, BluechiNodeContainer]):
+
+    node_foo = nodes[NODE_FOO]
+    config_files_agent = "config-files-agent"
+    config_file_location = "/var/tmp"
+    bluechi_agent_str = "bluechi-agent"
+    invalid_conf_str = "invalid.conf"
+    valid_conf_str = "valid.conf"
+
+    # Copying relevant config files into node container
+    for file in os.listdir(config_files_agent):
+        file_location = os.path.join(config_files_agent, file)
+        content = read_file(file_location)
+        node_foo.create_file(config_file_location, file_location, content)
+
+    LOGGER.info("Setting invalid.conf as conf file for foo and checking if failed")
+    restart_with_config_file(os.path.join(config_file_location, invalid_conf_str), node_foo, bluechi_agent_str)
+    assert node_foo.wait_for_unit_state_to_be(bluechi_agent_str, failed_status)
+
+    LOGGER.info("Setting valid.conf as conf file for node_foo and checking if active")
+    restart_with_config_file(os.path.join(config_file_location, valid_conf_str), node_foo, bluechi_agent_str)
```

Comment:
I'd check here only invalid config file and leaving checking valid config file for the port test, because for valid config file we should also check that configuration is applied